### PR TITLE
Staging -> Main

### PR DESCRIPTION
**Pull request by DeusTaNoComando**

### CODE REVIEW

File: .github/workflows/development.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
+name: CMS Plugin
```
Comment (DAzevedo26) [inline .github/workflows/development.yml:1]: notificaions

**Review by apoca — APPROVED** [review]

File: .husky/pre-commit [hunk 2]
```diff
@@ -0,0 +1 @@
+yarn format
```
Comment (DAzevedo26) [inline .husky/pre-commit:1]: missing commands?

File: src/mailer/dev-mailbox.ts [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+    handleMockEvent(
+        handler: (
+            handler: EmailEventHandler<string, any>,
+            event: EventWithContext,
+        ) => void,
+    ) {
+        this.handleMockEventFn = handler
+    }
+
```
Comment (DAzevedo26) [inline src/mailer/dev-mailbox.ts:96]: docs?

File: src/mailer/notification-processor.ts [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+        }
+        const transport = await this.getTransportSettings()
+        if (transport.type === 'file') {
+            // ensure the configured directory exists before
+            // we attempt to write files to it
+            const emailPath = transport.outputPath
+            await fs.ensureDir(emailPath)
+        }
+    }
+
```
Comment (DAzevedo26) [inline src/mailer/notification-processor.ts:63]: docs

File: .husky/pre-commit (hunk 2 above)
Comment (DeusTaNoComando) [reply .husky/pre-commit:1]: Just format for now, have to patch it later with the tests

File: codegen.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import type { CodegenConfig } from '@graphql-codegen/cli'
+
+const config: CodegenConfig = {
+    overwrite: true,
+    // This assumes your server is running on the standard port
+    // and with the default admin API path. Adjust accordingly.
+    schema: 'http://localhost:3000/admin-api',
```
[the remark this answers is not in the document]
Comment (DeusTaNoComando) [reply codegen.ts:7]: Yes, this file generates your codegen **_admin-types_** and **_shop-types_** for the plugin, so there are no problems for this to be like this :ok_hand:

**Review by DAzevedo26 — APPROVED** [review]

**Pull request merged**